app/util.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const request = require('request-promise');
+
+// returns a Date representing tomorrow at 12pm so that
+// we can issue darksky 'time machine' requests to get an
+// idea of the weather tomorrow at a time that should be
+// relevant to someone interested in the weather
+function getTomorrowAtNoon() {
+    let tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    tomorrow.setHours(12);
+    return tomorrow;
+}
+
+// fixed coordinates used to query darksky
+const latitude = 38.8892681;
+const longitude = -77.0501425;
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+const darkskyUrlBase = `https://api.darksky.net/forecast/${darkskyToken}/${latitude},${longitude}`;
+
+// returns a promise that resolves to the JSON representing the current weather
+// report from darksky on success
+function getCurrentWeatherReport() {
+    // we don't need the minutely or hourly forecast, so we'll exclude them
+    const darkskyCurrentlyUrl = `${darkskyUrlBase}` + '?exclude=minutely,hourly';
+    return request({ uri: darkskyCurrentlyUrl, json: true });
+}
+
+function getTomorrowsWeatherReport() {
+    const darkskyTomorrowUrl = `${darkskyUrlBase}` + getTomorrowAtNoon().getTime() + '?exclude=currently,minutely,hourly';
+    return request({ uri: darkskyTomorrowUrl, json: true });
+}
+
+function generateCurrentSummary(dataBlock) {
+    return generateWeatherReport(dataBlock, 'is currently', 'feels like');
+}
+
+function generateTomorrowsSummary(dataBlock) {
+    return generateWeatherReport(dataBlock, 'tomorrow will be', 'will feel like');
+}
+
+// maps the darksky 'icon' field to an appropriate emoji representation!
+// not supported by jshint yet - https://github.com/jshint/jshint/pull/2413
+const icon2Emoji = {
+    /* jshint ignore:start */
+    'clear-day': '\u{2600}',
+    'clear-night': '\u{1f31d}',
+    'rain': '\u{1f327}',
+    'snow': '\u{1f328}',
+    'sleet': '\u{2744}\u{2614}',
+    'wind': '\u{1f32c}',
+    'fog': '\u{1f32b}',
+    'cloudy': '\u{1f325}',
+    'partly-cloudy-day': '\u{1f324}',
+    'partly-cloudy-night': '\u{2601}'
+    /* jshint ignore:end */
+};
+
+// turns a darksky JSON data block into a human-readable string
+// https://darksky.net/dev/docs/response#data-block
+function generateWeatherReport(dataBlock, summaryVerb, temperatureVerb) {
+    let report = '';
+    let summary = dataBlock.summary.toLowerCase();
+
+    if (dataBlock.icon && icon2Emoji.hasOwnProperty(dataBlock.icon)) {
+        report = icon2Emoji[dataBlock.icon] + ' ';
```

Comment:
Changed this to build up the string pieces in an array and then join them at the end.